examples/axis-rotate-auto/index.js

```diff
@@ -0,0 +1,82 @@
+const domain = [
+    'Carrots',
+    'Bananas',
+    'Sausages',
+    'Pickles',
+    'Aubergines',
+    'Artichokes',
+    'Spinach',
+    'Cucumber'
+];
+
+const draw = () => {
+    const renderAxis = (axisFactory, selector) => {
+        const container = document.querySelector(selector);
+
+        const scale = d3.scaleBand().domain(domain);
+        const axis = axisFactory(scale);
+
+        d3.select(container)
+            .on('draw', () => {
+                d3.select(container)
+                    .select('svg')
+                    .call(axis);
+            })
+            .on('measure', () => {
+                const { width, height } = event.detail;
+                const side = container.getAttribute('class');
+
+                const vertical = side === 'left' || side === 'right';
+                scale.range([0, vertical ? height : width]);
+
+                const topOffset = side === 'top' ? 80 : 0;
+                const leftOffset = side === 'left' ? 80 : 0;
+
+                d3.select(container)
+                    .select('svg')
+                    .attr(
+                        'viewBox',
+                        `${-leftOffset} ${-topOffset} ${width} ${height}`
+                    );
+            });
+        container.requestRedraw();
+    };
+
+    renderAxis(
+        scale => fc.axisLabelRotate(fc.axisOrdinalTop(scale)),
+        '#topAuto'
+    );
+    renderAxis(
+        scale => fc.axisLabelRotate(fc.axisOrdinalBottom(scale)),
+        '#bottomAuto'
+    );
+    renderAxis(
+        scale => fc.axisLabelRotate(fc.axisOrdinalLeft(scale)),
+        '#leftAuto'
+    );
+    renderAxis(
+        scale => fc.axisLabelRotate(fc.axisOrdinalRight(scale)),
+        '#rightAuto'
+    );
+
+    renderAxis(
+        scale => fc.axisLabelRotate(fc.axisOrdinalTop(scale)).labelRotate(30),
+        '#topFixed'
+    );
+    renderAxis(
+        scale =>
+            fc.axisLabelRotate(fc.axisOrdinalBottom(scale)).labelRotate(30),
+        '#bottomFixed'
+    );
+    renderAxis(
+        scale => fc.axisLabelRotate(fc.axisOrdinalLeft(scale)).labelRotate(30),
+        '#leftFixed'
+    );
+    renderAxis(
+        scale => fc.axisLabelRotate(fc.axisOrdinalRight(scale)).labelRotate(30),
+        '#rightFixed'
+    );
+};
+
+draw();
+window.addEventListener('resize', () => draw());
```

Comment:
Just spotted this line. If you wrap all of the `d3fc-svg`s in a `d3fc-group` (i.e. 1 x `d3fc-group` with n x `d3fc-svg` children) with an `auto-resize` attribute, it should handle the redrawing for you so you can remove the `draw` function and just make the code inline.